### ISSUE TITLE
(andydavis) Find cleaner way to map each grad output returned by 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#69]
gradient function to the src node/output to which it should be
backproped. Maybe grad functions can return a vector of Output pairs to
make this association explicit.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [23441bf71dbd0a10bcace160165ce01bf6deac9a](https://github.com/imdone/tensorflow/commit/23441bf71dbd0a10bcace160165ce01bf6deac9a) by A. Unique TensorFlower gardener@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/cc/framework/gradients.cc#L520-L528
